test/appSession.tests.js

```diff
@@ -54,19 +55,11 @@ describe('appSession', function() {
 
   describe('existing session cookies', () => {
     const appSessionMw = appSession(defaultConfig);
-    const thisReq = {
-      // Encypted '{sub:"__test_sub__"}' with '__test_secret__'
-      get: () => 'identity=eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIiwiemlwIjoiREVGIiwidWF0IjoxNTc3ODI2NzY5' +
-      'LCJpYXQiOjE1Nzc4MjY3NjksImV4cCI6MTU3ODQzMTU2OX0..4XocWueShMw1cD_b.EhS_rNI4HeCFSlJTxKowE1SwLfsEfg' +
-      '.JKMnZOkBjwi-9Z5BSHliiw'
-    };
-    const result = appSessionMw(thisReq, res, next);
-
-    it('should call next', function() {
-      assert.ok(result);
-    });
+    const thisReq = {get: () => 'identity=' + sessionEncryption.encrypted};
 
     it('should set the identity on req', function() {
+      const result = appSessionMw(thisReq, res, next);
+      assert.ok(result);
```

Comment:
If the line above is async (see `next` is being passed), how can you ensure the result will be set by when this line is reached? Unless I'm missing something

Reply:
The `next()` there isn't a good indication of whether that function is async or not. You would see `await` in front if it was.